CSharpSourceCode/Abilities/Ability.cs

```diff
@@ -93,10 +96,7 @@ protected virtual void DoCast(Agent casterAgent)
                 IsCasting = true;
                 var timer = new Timer(Template.CastTime * 1000);
                 timer.AutoReset = false;
-                timer.Elapsed += (s, e) =>
```

Comment:
That's actually my change. It was required to get rid of this timer because it led to memory access violation crashes. New way is always called from Tick() and not from a Timer delegate which ensures it always happens on the main thread.